frontend/src/hooks/useTransactionListener.ts

```diff
@@ -13,8 +14,24 @@ export function useTransactionListener() {
     );
   }
 
+  async function getTransaction(hash: string, requestTime: number) {
+    const result = await transactionsStore.getTransaction(hash);
+
+    // Only process the result if this is still the latest request for this hash
+    if (latestRequestTimes.get(hash) === requestTime) {
```

Comment:
@denishacquin I'm not sure that I understand this. Could you please clarify? it seems a bit odd

Reply:
@cristiam86 Let's park this PR for now since it should be solved in the backend by an open PR of @kstroobants .

As soon as we merged said PR, let's see if the symptoms still occur and if this fix makes sense still. Moving back to draft.